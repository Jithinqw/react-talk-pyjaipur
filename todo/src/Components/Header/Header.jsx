import React from 'react';
import { Navbar, Nav } from 'react-bootstrap';
import history from '../../history';
import { removeToken } from '../../Utils/tokenUtil';

export default class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            loggedIn: false,
        }
    }

    componentDidMount () {
        if(history.location.pathname === "/dashboard") {
            this.setState({loggedIn: true});
        }
    }
    
    logout = () => {
        removeToken('token');
        history.push('/login');
    }

    renderRegister() {
        if(history.location.pathname === "/register") {
            return <Nav.Link href="/login" className="loginBtn">Login</Nav.Link>
        } else {
            return null;
        }
    }

    renderLogin() {
        if(history.location.pathname === "/login") {
            return <Nav.Link href="/register" className="registerBtn">Register</Nav.Link>
        } else {
            return null;
        }
    }

    render() {
        return(
            <Navbar bg="light" expand="lg">
                <Navbar.Brand href="/dashboard">The Todo App</Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav"></Navbar.Toggle>
                <Navbar.Collapse id="basic-navbar-nav"></Navbar.Collapse>
                <Nav>
                    {
                        this.renderRegister()
                    }
                    {
                        this.renderLogin()
                    }
                    {
                        this.state.loggedIn ? <Nav.Link href="#" onClick={()=>this.logout()}>Logout</Nav.Link> : null
                    }
                </Nav>
            </Navbar>
        )
    }
}