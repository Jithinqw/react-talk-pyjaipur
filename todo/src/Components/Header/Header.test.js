import React from 'react'
import Header from './Header';
import { Navbar } from 'react-bootstrap';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Unit Test for the Header', ()=> {

    it('Check if Header is rendering', ()=>{
       const HeaderMount = mount(<Header />);
       expect(HeaderMount.find(<Navbar.Brand href="/dashboard">The Todo App</Navbar.Brand>).contains('The Todo App'));
    })
})