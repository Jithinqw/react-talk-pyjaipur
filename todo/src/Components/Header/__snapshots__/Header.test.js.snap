// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test for the Header Check if Header is rendering 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <a
    className="navbar-brand"
    href="/dashboard"
  >
    The Todo App
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  />
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  />
</nav>
`;
